### PR TITLE
Enhance the contracts list by removing the duplicate requests, and reduce the time limit when loading the contracts by removing the `onMounted` lifecycle

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

Enhance the contracts list by removing the duplicate requests, and reduce the time limit when loading the contracts by removing the `onMounted` lifecycle

### Changes

- Removed the 'onMount' from the 'onMounted' life cycle since it is not needed because we call the 'loadContracts' function on each change that happens on the table.
- Fix the contracts table takes too much time to load.
- Fix the undefined value of the total cost.

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2802
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2798
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2889
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2800

### Screenshots/Video
Screenshot from 2024-06-04 18-52-10

 [Screencast from 06-04-2024 06:46:01 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/1ae65b10-66d1-4666-860a-b3d662092af4)

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by amiraabouhadid — COMMENTED** [review]
Good job, would be nice though if we can make only one request for the same node used by multiple contracts instead of for example requesting node 11 here twice 
Screenshot from 2024-06-05 10-50-44

**Review by zaelgohary — APPROVED** [review]

**Review by maayarosama — APPROVED** [review]

**Comment by 0oM4R** [comment]
changing table options still takes so much time to load, is there anyway to enhance it ? like client side pagination, for example ?

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 1]
```diff
@@ -348,6 +348,12 @@ async function loadContracts(options: { page: number; itemsPerPage: number; cont
   }
 }
 
+async function _getTotalCost() {
+  totalCost.value = getTotalCost(contracts.value);
+  const TFTInUSD = await queryClient.tftPrice.get();
```
Comment (MohamedElmdary) [inline packages/playground/src/weblets/tf_contracts_list.vue:353]: await might throw error please handle any possible error

**Review by MohamedElmdary — APPROVED** [review]

**Comment by maayarosama** [comment]
Good job

**Review by maayarosama — APPROVED** [review]

**Pull request merged**